### PR TITLE
Support to Bluetooth RPC Host for nRF54H20

### PR DESCRIPTION
**Pull request by ppryga-nordic**
There is a possibility to use complete Bluetooth stack on e.g. nRF5340 network core, meaning Host and Controller run in same core. An application communicates with the stack over Bluetooth RPC subsystem that wraps Bluetooth API into RPC.

The PR adds possibility to use the same approach for samples running on nRF54H20 SOC.
To make it easy, the nordic-rcp-host snippet was added. 
Also a peripheral_uart Kconfig.sysbuild was modified to make possible to select radio domain target sample application.

One additional change was done to Bluetooth RPC subsystem, handling of `settings` initialization on remote. That change was done because former implementation required to increase main thread stack size.

### CODE REVIEW

File: snippets/nordic-rpc-host/snippet.yml
Comment (ppryga-nordic) [reply snippets/nordic-rpc-host/snippet.yml:13]: :/ Sorry forgotten to add it to commit

File: snippets/nordic-rpc-host/boards/nrf54h20dk_nrf54h20_cpurad.overlay
Comment (ppryga-nordic) [reply snippets/nordic-rpc-host/boards/nrf54h20dk_nrf54h20_cpurad.overlay:1]: Agree, the part about move of the existing partitions.

Comment (ppryga-nordic) [reply snippets/nordic-rpc-host/boards/nrf54h20dk_nrf54h20_cpurad.overlay:1]: `nrf54h20dk_nrf54h20-mem-map-move.dtsi` is it fine if I use for common file?

**Comment by ppryga-nordic** [comment]
Fixed licenses and added CODEOWNERS entry for the new snippet.

File: snippets/nordic-rpc-host/snippet.yml
Comment (ppryga-nordic) [reply snippets/nordic-rpc-host/snippet.yml:13]: Done

File: snippets/nordic-rpc-host/README.rst
Comment (ppryga-nordic) [reply snippets/nordic-rpc-host/README.rst]: Done

File: snippets/nordic-rpc-host/boards/nrf54h20dk_nrf54h20_cpurad.overlay
Comment (ppryga-nordic) [reply snippets/nordic-rpc-host/boards/nrf54h20dk_nrf54h20_cpurad.overlay:1]: Done - common file names `nrf54h20dk_nrf54h20-mem-map-move.dtsi`

File: snippets/nordic-rpc-host/boards/nrf54h20dk_nrf54h20_cpurad.overlay
Comment (hubertmis) [inline snippets/nordic-rpc-host/boards/nrf54h20dk_nrf54h20_cpurad.overlay:9]: I would move this entry to `nrf54h20dk_nrf54h20-mem-map-move.dtsi` as well. Leaving here only the `status = "okay";` property.

File: snippets/nordic-rpc-host/boards/nrf54h20dk_nrf54h20_cpurad.overlay
Comment (hubertmis) [reply snippets/nordic-rpc-host/boards/nrf54h20dk_nrf54h20_cpurad.overlay:1]: The new file seems to be missing in this commit

File: snippets/nordic-rpc-host/boards/nrf54h20dk_nrf54h20_cpurad.overlay
Comment (ppryga-nordic) [reply snippets/nordic-rpc-host/boards/nrf54h20dk_nrf54h20_cpurad.overlay:9]: Moved, I left `status` and `zephyr,settings_partition`.

**Review by hubertmis — APPROVED** [review]
Looks good!

File: samples/bluetooth/peripheral_uart/Kconfig.sysbuild
Comment (dchat-nordic) [inline samples/bluetooth/peripheral_uart/Kconfig.sysbuild:18]:
```suggestion
	default y if !NRF_IPC_RADIO_RPC_HOST
```
This config enables the ipc_radio firmware on network core, so if you want to use rpc_host image as network core this option needs to be disabled.

**Review by dchat-nordic — COMMENTED** [review]
The code itself looks good, but according to NCSDK-27179 the default firmware for network core should be ipc_radio.
The ipc_radio supports the rpc configuration, have you tried to use it instead of rpc_host sample?

File: snippets/nordic-rpc-host/boards/nrf54h20dk_nrf54h20_cpurad.overlay
Comment (nordicjm) [inline snippets/nordic-rpc-host/boards/nrf54h20dk_nrf54h20_cpurad.overlay:12]: too many indents

File: samples/bluetooth/peripheral_uart/Kconfig.sysbuild
Comment (nordicjm) [inline samples/bluetooth/peripheral_uart/Kconfig.sysbuild:15]: why do we need yet another kconfig for the same thing?

Comment (ppryga-nordic) [reply samples/bluetooth/peripheral_uart/Kconfig.sysbuild:15]: That was my bad, I should have added description to `NRF_DEFAULT_RPC_HOST`.
Anyway the config was removed. Since there is change in approach and `ipc_radio` application will be used instead of `rpc_host` part of the content in this file was extracted to `sysbuild.conf` and `sysbuild_bt_rpc.conf`.

File: snippets/nordic-rpc-host/boards/nrf54h20dk_nrf54h20_cpurad.overlay
Comment (ppryga-nordic) [reply snippets/nordic-rpc-host/boards/nrf54h20dk_nrf54h20_cpurad.overlay]: Fixed

File: samples/bluetooth/peripheral_uart/Kconfig.sysbuild
Comment (ppryga-nordic) [reply samples/bluetooth/peripheral_uart/Kconfig.sysbuild:10]: After offline talk the sample was adjusted to use ipc_radio application for Radio domain + Bluetooth RPC_HOST subsystem.

**Review by dchat-nordic — APPROVED** [review]

**Comment by ppryga-nordic** [comment]
changed `sample.yaml` to workaround twister issue: missing `CONFIG_` at hte beginning of entries for `extra_configs`.

**Comment by ppryga-nordic** [comment]
PR is blocked by failing tests that require update in SDC library.

**Review by nordicjm — APPROVED** [review]

**Comment by guwa** [comment]
BLE CI has a setup failure due to commit message check. It should not block the PR as long as the test job has passed

**Pull request merged**
